embedded/imu_driver/imu_balsaq.cc

```diff
@@ -62,9 +64,13 @@ void ImuBq::loop() {
     ImuMessage msg;
     msg.timestamp = get_current_time();
 
-    const jcc::Vec3 accel_mpss = imu.driver.read_accel_mpss();
-    const jcc::Vec3 angvel_radps = imu.driver.read_gyro_radps();
-    const jcc::Vec3 mag_utesla = imu.driver.read_magnetometer_utesla();
+    jcc::Vec3 accel_mpss = {};
+    jcc::Vec3 angvel_radps = {};
+    jcc::Vec3 mag_utesla = {};
+
+    if (!imu.driver.read_agm(accel_mpss, angvel_radps, mag_utesla)) {
```

Comment:
Use `third_party/experiments/out.hh`

if (!imu.driver.read_agm(out(accel_mpss), out(angvel_radps), out(mag_utesla))) {

But strongly prefer to return a struct or optional struct instead of using output arguments -- I would do that instead